src/lfortran/semantics/ast_common_visitor.h

```diff
@@ -883,13 +883,56 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
                             } else if(sa->m_attr == AST::simple_attributeType
                                     ::AttrIntrinsic) {
                                 // Ignore Intrinsic attribute
+                            } else if (sa->m_attr == AST::simple_attributeType
+                                    ::AttrExternal) {
+                                // TODO
+                                throw SemanticError("Attribute declaration not "
+                                    "supported yet", x.base.base.loc);
                             } else {
                                 throw SemanticError("Attribute declaration not "
                                         "supported", x.base.base.loc);
                             }
                         }
                     }
                 }
+            // enable sole `dimension` attribute
+            } else if (AST::is_a<AST::AttrDimension_t>(*x.m_attributes[0]) && x.n_attributes == 1 ){   
+            
+                for (size_t i=0;i<x.n_syms;++i) { // symbols for line only
+                    AST::var_sym_t &s = x.m_syms[i];
+                    std::string sym = to_lower(s.m_name);
+                    ASR::symbol_t *get_sym = current_scope->get_symbol(sym);
+                    // get actual variable
+                    if (get_sym == nullptr) throw SemanticError("Variable undeclared", x.base.base.loc);
+                    // TODO: check if we can abuse dimension to just update a variables dimensions
+
+                    if (ASR::is_a<ASR::Variable_t>(*get_sym)) {
+                        auto v = (ASR::Variable_t*)get_sym;
+                        if (ASR::is_a<ASR::Integer_t>(*(v->m_type))) {
+                            auto actual = (ASR::Integer_t*)(v->m_type);
+                            Vec<ASR::dimension_t> dims;
+                            dims.reserve(al, 0);
+                            process_dims(al, dims, x.m_syms[i].m_dim, x.m_syms[i].n_dim);
+                            v->m_type = (ASR::ttype_t*)LFortran::ASR::make_Integer_t(al, actual->base.base.loc, actual->m_kind, dims.data(), x.m_syms[i].n_dim);
```

Comment:
ASR is constructed to never reuse nodes, so that they can be modified in place:
```suggestion
                            actual->m_dims = dims.data();
                            actual->n_dims = dims.size();
```